base/multi.jl

```diff
@@ -1060,7 +1060,18 @@ function process_tcp_streams(r_stream::TCPSocket, w_stream::TCPSocket, incoming)
     message_handler_loop(r_stream, w_stream, incoming)
 end
 
-function process_messages(r_stream::IO, w_stream::IO, incoming=true)
+"""
+    Base.process_messages(r_stream::IO, w_stream::IO, incoming::Bool=true)
+
+Called by cluster managers using custom transports. It should be called when the custom
+transport implementation receives the first message from a remote worker. The custom
+transport must manage a logical connection to the remote worker and provide two
+`IO` objects, one for incoming messages and the other for messages addressed to the
+remote worker.
+If `incoming` is `true`, schedules reads from `r_stream` and writes to `w_stream`.
+Otherwise, schedules in the reverse direction.
```

Comment:
This statement is not correct. `incoming` only refers to which side of the connected pair initiated the connection. If `true`, the remote peer initiated the connection. This is used as part of the initial connection authentication handshake. The connection initiator side writes both the cookie and version before any Julia messages. The other side only the version.

`r_stream` and `w_stream` are always read from and written to respectively.